pkg/controller/core/clusterqueue_controller_test.go

```diff
@@ -0,0 +1,96 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
+)
+
+func TestUpdateCQStatus(t *testing.T) {
+	testCases := map[string]struct {
+		cqConditions     []metav1.Condition
+		conditionStatus  metav1.ConditionStatus
+		reason, msg      string
+		wantCqConditions []metav1.Condition
+	}{
+		"empty conditions": {
+			cqConditions:    nil,
+			conditionStatus: metav1.ConditionFalse,
+			reason:          "FlavorNotFound",
+			msg:             "Can't admit new workloads; some flavors are not found",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "FlavorNotFound",
+				Message: "Can't admit new workloads; some flavors are not found",
+			}},
+		},
+		"same condition status": {
+			cqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionTrue,
+				Reason:  "Ready",
+				Message: "Can admit new workloads",
+			},
+			},
+			conditionStatus: metav1.ConditionTrue,
+			reason:          "Ready",
+			msg:             "Can admit new workloads",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionTrue,
+				Reason:  "Ready",
+				Message: "Can admit new workloads",
+			}},
+		},
+		"same condition status with different reason and message": {
+			cqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "FlavorNotFound",
+				Message: "Can't admit new workloads; Can't admit new workloads; some flavors are not found",
+			}},
+			conditionStatus: metav1.ConditionFalse,
+			reason:          "Terminating",
+			msg:             "Can't admit new workloads; clusterQueue is terminating",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "Terminating",
+				Message: "Can't admit new workloads; clusterQueue is terminating",
+			}},
+		},
+		"different condition status": {
+			cqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "FlavorNotFound",
+				Message: "Can't admit new workloads; some flavors are not found",
+			}},
+			conditionStatus: metav1.ConditionTrue,
+			reason:          "Ready",
+			msg:             "Can admit new workloads",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionTrue,
+				Reason:  "Ready",
+				Message: "Can admit new workloads",
+			}},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			cqStatus := &kueue.ClusterQueueStatus{Conditions: tc.cqConditions}
+			UpdateCqStatus(cqStatus, 0, 0, kueue.UsedResources{}, tc.conditionStatus, tc.reason, tc.msg)
```

Comment:
I think you also need to check the bool value returned by `UpdateCqStatus` to help to test the `DeepEqual` is available?

Reply:
Makes sense.
I will add the logic to verify that value returned by `UpdateCqStatus`.
Thanks!